core/jobs/batch_jobs/firebase_validation_jobs.py

```
@@ -0,0 +1,148 @@
+
+        Args:
+            grouped: RecordsGroupedByEmail. The grouped records.
+
+        Yields:
+            pvalue.TaggedOutput. An OK tag with ok count or ERR tag with reason.
+        """
+        oppia_records = list(grouped['weak_entries'])
+        firebase_records = list(grouped['strong_entries'])
+        collisions_found = False
+
+        if len(ids := sorted(set(r.auth_id for r in firebase_records))) > 1:
```

Ruff (0.15.1)

[convention] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)